From 1189cd6f19bc562d5009a7b566ca5e8cb7a6906a Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@hmsreliant.think-freely.org>
Date: Fri, 23 Mar 2012 15:02:30 -0400
Subject: [PATCH] numa: only ignore the unspecified node during placement when
 numa is available

Serj Kalichev reported a problem with irqbalance on non-numa systems, indicating
that irqs were not balanced because the placement algorithm ignores the
unspecified node.  We need to not do that, but only in cases where we don't have
any other nodes.  If we have a valid numa system, then we want to always choose
one of the real nodes over the unspecified one, but in non-numa systems we can
and should allow the selection of the unspecified (and only) node

Signed-off-by: Neil Horman <nhorman@tuxdriver.com>
Reported-by: Serj Kalichev <serj.kalichev@gmail.com>
---
 placement.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/placement.c b/placement.c
index 108ccc9..a1e33f3 100644
--- a/placement.c
+++ b/placement.c
@@ -48,7 +48,7 @@ static void find_best_object(struct topo_obj *d, void *data)
 	/*
  	 * Don't consider the unspecified numa node here
  	 */
-	if ((d->obj_type == OBJ_TYPE_NODE) && (d->number == -1))
+	if (numa_avail && (d->obj_type == OBJ_TYPE_NODE) && (d->number == -1))
 		return;
 
 	/*
-- 
1.7.11.4

